lib/std/math/Interval.zig

```diff
@@ -0,0 +1,110 @@
+const std = @import("../std.zig");
+
+/// Create an interval of type T.
+/// See https://en.wikipedia.org/wiki/Interval_(mathematics) for more info.
+pub fn Interval(comptime T: type) type {
```

Comment:
What if open/closed-ness was a comptime property of the type?

Reply:
because its defined in the `Endpoint` struct and its not something that inherently needs to be an invariant so I left it there in the struct over having an `.init()` function